lib/semian/lru_cache.rb

```diff
@@ -0,0 +1,51 @@
+class LRUCache
+  extend Forwardable
+  def_delegators :@table, :size, :count
+  def_delegators :@table, :empty?
+  def_delegators :@table, :keys
+  def_delegators :@table, :clear
+
+  attr_accessor :max_items, :table
+  def initialize(max_items)
+    @max_items = max_items
+    @table = {}
+  end
+
+  def set(key, value)
+    @table.delete(key)
+    @table[key] = value
+    clear_unused_resources
+    # @table.shift if @table.length > @max_items
+    value
+  end
+
+  def get(key)
+    if found = @table.delete(key)
+      @table[key] = found
+    else
+      nil
+    end
+  end
+
+  def []=(key, value)
+    set(key, value)
+  end
+
+  def [](key)
+    get(key)
+  end
+
+  def delete(key)
+    @table.delete(key)
+  end
+
+  private
+
+  def clear_unused_resources
+    @table.each do |key, value|
+      next unless value.instance_of?(Semian::ProtectedResource)
```

Comment:
When would it ever _not_ be a protected resource?

Reply:
I think I added this line when I was running some tests and forgot to remove it  🤦‍♂️

Reply:
Alright so the problem is with this test: https://github.com/Shopify/semian/blob/master/test/resource_test.rb#L80

I was under the impression it would be possible to use a `Resource` Do you think I could get a better fix than verifying the instance?